File: Ex3.java
package Ex3;

class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    public String getTitular() {
        return titular;
    }

    public double getSaldo() {
        return saldo;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.println("Depósito de R$" + valor + " realizado com sucesso!");
        } else {
            System.out.println("Valor de depósito inválido.");
        }
    }

    public void sacar(double valor) {
        if (valor <= saldo) {
            saldo -= valor;
            System.out.println("Saque de R$" + valor + " realizado com sucesso!");
        } else {
            System.out.println("Saldo insuficiente para o saque de R$" + valor + ".");
        }
    }

    public void exibirSaldo() {
        System.out.println("Titular: " + titular + ", Saldo atual: R$" + saldo);
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("João", 1000.00);
        
        conta.exibirSaldo();
        
        conta.depositar(500.00);
        conta.exibirSaldo();
        
        conta.sacar(300.00);
        conta.exibirSaldo();
        
        conta.sacar(1500.00);
    }
}

File: ex3.go
package main

import (
	"fmt"
)

type ContaBancaria struct {
	titular string
	saldo   float64
}

func (c *ContaBancaria) Depositar(valor float64) {
	if valor > 0 {
		c.saldo += valor
		fmt.Printf("Depósito de R$ %.2f realizado. Novo saldo: R$ %.2f\n", valor, c.saldo)
	} else {
		fmt.Println("Valor de depósito deve ser maior que zero.")
	}
}

func (c *ContaBancaria) Sacar(valor float64) {
	if valor > 0 && valor <= c.saldo {
		c.saldo -= valor
		fmt.Printf("Saque de R$ %.2f realizado. Saldo restante: R$ %.2f\n", valor, c.saldo)
	} else {
		fmt.Println("Saque não realizado. Verifique se o valor é positivo e não ultrapassa o saldo.")
	}
}

func (c ContaBancaria) ExibirSaldo() {
	fmt.Printf("Saldo da conta de %s: R$ %.2f\n", c.titular, c.saldo)
}

func main() {
	conta1 := ContaBancaria{titular: "João", saldo: 1000.0}
	conta1.ExibirSaldo()

	conta1.Depositar(500.0)
	conta1.Sacar(200.0)
	conta1.ExibirSaldo()

	conta1.Sacar(1500.0)
	conta1.Depositar(-100.0)
}
